File: Day_2/main.cpp
#include <iostream>
#include <fstream>
#include <string>

int previous(int n)
{
    if (n == 1)
        return 3;
    else
        return n - 1;
}

int next(int n)
{
    if (n == 3)
        return 1;
    else
        return n + 1;
}

int rps1(char p1, char p2)
{
    int s1 = (int)(p1 - 'A') + 1;
    int s2 = (int)(p2 - 'X') + 1;

    if (s1 == s2)
        return (s2 + 3);
    else if (s1 == previous(s2))
        return (s2 + 6);
    return (s2);
}

int rps2(char p1, char p2)
{
    int s1 = (int)(p1 - 'A' + 1);
    int s2 = (int)(p2 - 'X' + 1);

    if (s2 == 1)
        return (previous(s1));
    else if (s2 == 2)
        return (s1 + 3);
    else if (s2 == 3)
        return (next(s1) + 6);
    return (0);
}

int main()
{
    int score1 = 0;
    int score2 = 0;
    std::ifstream myfile ("input.txt");
    if (myfile.is_open())
    {
        std::string line;
        while ( getline (myfile,line) )
        {
            score1 += rps1(line[0], line[2]);
            score2 += rps2(line[0], line[2]);
        }
    }
    std::cout << "Points for the first game : " << score1 << std::endl;
    std::cout << "Points for the second game : " << score2 << std::endl;
    return (0);
}

File: Day_2/main.py
def next(n):
    if n == 3:
        return 1
    return n + 1

def previous(n):
    if n == 1:
        return 3
    return n - 1

def rps1(p1, p2):
    p1 = ord(p1) - 64
    p2 = ord(p2) - 87
    if p1 == p2:
        return p2 + 3
    if p1 == previous(p2):
        return 6 + p2
    return p2

def rps2(p1, p2):
    p1 = ord(p1) - 64
    p2 = ord(p2) - 87
    if p2 == 1:
        return previous(p1)
    if p2 == 2:
        return 3 + p1
    return 6 + next(p1)


res1 = int(0)
res2 = int(0)
with open("input.txt") as f:
    lines = f.readlines()
    for line in lines:
        res1 += rps1(line[0], line[2])
        res2 += rps2(line[0], line[2])
    print("Points for the first game : " + str(res1))
    print("Points for the second game : " + str(res2))
        